src/rendering/renderers/gl/context/GlContextSystem.ts

```diff
@@ -197,28 +209,45 @@ export class GlContextSystem implements System<ContextSystemOptions>
      * Initialize from context options
      * @protected
      * @see https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext
+     * @param preferWebGLVersion
      * @param {object} options - context attributes
      */
-    protected initFromOptions(options: WebGLContextAttributes): void
+    protected initFromOptions(preferWebGLVersion: 1 | 2, options: WebGLContextAttributes): void
     {
-        const gl = this.createContext(this._renderer.view.canvas, options);
+        const gl = this.createContext(this._renderer.view.canvas, preferWebGLVersion, options);
 
         this.initFromContext(gl);
     }
 
     /**
      * Helper class to create a WebGL Context
      * @param canvas - the canvas element that we will get the context from
+     * @param preferWebGLVersion
      * @param options - An options object that gets passed in to the canvas element containing the
      *    context attributes
      * @see https://developer.mozilla.org/en/docs/Web/API/HTMLCanvasElement/getContext
      * @returns {WebGLRenderingContext} the WebGL context
      */
-    public createContext(canvas: ICanvas, options: WebGLContextAttributes): GlRenderingContext
+    public createContext(canvas: ICanvas, preferWebGLVersion: 1 | 2, options: WebGLContextAttributes): GlRenderingContext
     {
-        const gl = canvas.getContext('webgl2', options);
+        let gl: WebGL2RenderingContext | WebGLRenderingContext;
+
+        if (preferWebGLVersion === 2)
+        {
+            gl = canvas.getContext('webgl2', options);
+        }
+
+        if (!gl)
+        {
+            gl = canvas.getContext('webgl', options);
+
+            if (!gl)
+            {
+                // fail, not able to get a context
+                throw new Error('This browser does not support WebGL. Try using the canvas renderer');
+            }
+        }
 
-        this.webGLVersion = 2;
```

Comment:
you are no longer setting the `webGlVersion` at all

shouldn't this be set to the `preferWebGLVersion`?

Reply:
yo! This gets set in line 194:
```this.webGLVersion = gl instanceof WebGL2RenderingContext ? 2 : 1;```

stuck it there as the gl context may have been provided and we need to set it based in that.